lib/transport.js

```diff
@@ -0,0 +1,29 @@
+'use strict'
+
+const { createRequire } = require('module')
+const caller = require('get-caller-file')
+const { isAbsolute } = require('path')
+
+const ThreadStream = require('thread-stream')
+
+function transport (filename, workerData) {
+  if (!(isAbsolute(filename) || filename.indexOf('file://') === 0)) {
```

Comment:
also support declaring transports per log level 

`transport({ error: 'transport-a', info: 'transport-b'})` this needs more thought on how to support various configs (e.g. error and below etc)

Reply:
That's PR #1004.

Reply:
yeah, but I think we can make it easier by using multistream internally here. It can still be provided separately (as is current), but we can improve this API so that it's opaque for most cases

Reply:
I'd also like to see something like:

`pino(pino.transport(config), pino.transport(anotherConfig))`

which allows for multiple worker threads 

So with multistream, plus the above there's ultimate flexibility

Reply:
I think this could open the door for intense - say trace log - processing in a dedicated worker thread while another worker thread could handle the usual use cases

Reply:
I think the best thing we could do is to support:

```
var streams = [
  { level: 'info', destination: 'stdout' },
  { level: 'warn', module: 'pino-socket', opts: { ... } },
  { level: 'error', module: 'pino-elasticsearch', opts: { ... } }
]

pino.transport(streams)
```

Which will create a new thread with all those transports configured.

Reply:
what about:

```js
pino({
  info: pino.destination(1),
  warn: pino.transport('pino-socket'),
  error: [pino.transport('pino-socket'), pino.transport('pino-elasticsearch')]
})
```

Reply:
I know there's logistical issues here but I'm thinking in terms of nicest API first and working backwards

Reply:
This solves the logistics and makes it nicer - `pino.transport` can return a "transport group" - which is a function that registers transports to run in the same worker thread (hence group):

```js
const transport = pino.transport()
pino({
  info: 1,
  warn: transport('pino-socket', opts),
  error: [transport('pino-socket', opts), transport('pino-elasticsearch', moreOpts)]
})
```

Reply:
For putting levels together:

```
const transport = pino.transport()
pino({
  '>=30': 1,
  20: transport('pino-socket', opts),
  '<20': [transport('pino-socket', opts), transport('pino-elasticsearch', moreOpts)]
})
```

Reply:
The beauty here is also that if you pass arguments to `pino.transport` it can work as currently defined - return a stream that can be passed to `pino` like pino.destination does

Reply:
@mcollina thoughts? :point_up:

Reply:
fair enough. - it also matches how pino.destination works